LibreNMS/Plugins/Plugin.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace LibreNMS\Plugins;
+
+use View;
+
+abstract class Plugin
+{
+    public function __construct()
+    {
+        View::addLocation(base_path('html/plugins/' . self::className() . '/ressources/views'));
+    }
+
+    final protected static function viewPath()
+    {
+        return debug_backtrace()[1]['function'];
+    }
+
+    final protected static function className()
+    {
+        return str_replace(__NAMESPACE__ . '\\', '', get_called_class());
+    }
+
+    public function menu()
+    {
+        $name = self::className();
+        echo view(self::viewPath(), compact('name'));
```

Comment:
instead of using compact, just make an array:
```php
echo view(self::viewPath(), ['name' => self::className()]);
```

(and other places too)

Reply:
The compact() function is extremely unintuitive and people generally don't know what it is doing.  I try to avoid it in general.

Reply:
ok, after i changed it i totally agree with you ;-)